tests/system/python/packages/test_authentication.py

```diff
@@ -0,0 +1,2211 @@
+                     headers={"authorization": PASSWORD_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': 'Password has been updated successfully for user id:<3>'} == jdoc
+
+    def test_update_password_with_certificate_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", "/foglamp/user/any2/password", body=json.dumps({"current_password": "User@123",
+                                                                            "new_password": "F0gl@mp1"}),
+                     headers={"authorization": CERT_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': 'Password has been updated successfully for user id:<5>'} == jdoc
+
+    def test_reset_user_with_password_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", "/foglamp/admin/3/reset", body=json.dumps({"role_id": 1, "password": "F0gl@mp!"}),
+                     headers={"authorization": PASSWORD_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': 'User with id:<3> has been updated successfully'} == jdoc
+
+    def test_reset_user_with_certificate_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", "/foglamp/admin/5/reset", body=json.dumps({"role_id": 1, "password": "F0gl@mp!"}),
+                     headers={"authorization": CERT_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': 'User with id:<5> has been updated successfully'} == jdoc
+
+    def test_delete_user_with_password_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("DELETE", "/foglamp/admin/4/delete", headers={"authorization": PASSWORD_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': "User has been deleted successfully"} == jdoc
+
+    def test_delete_user_with_certificate_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("DELETE", "/foglamp/admin/6/delete", headers={"authorization": CERT_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': "User has been deleted successfully"} == jdoc
+
+    def test_logout_all_with_password_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
```

Comment:
add test snippet that loggedout user is not able to access something (except ping) with old token